### PR TITLE
Move the azure-spring-cloud-starter-appconfiguration-config module from sdk/spring to sdk/appconfiguration

### PR DESCRIPTION
**Pull request by jacko9et**
Move the azure-spring-cloud-starter-appconfiguration-config module from sdk/spring to sdk/appconfiguration

### CODE REVIEW

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-appconfiguration-sample/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Spring Cloud Azure Config Sample client library for Java
 
-This sample describes how to use [spring-cloud-starter-azure-appconfiguration-config](../../azure-spring-cloud-starter-appconfiguration-config/) to load configuration properties from Azure Configuration Service to Spring Environment.
+This sample describes how to use [spring-cloud-starter-azure-appconfiguration-config](../../../appconfiguration/azure-spring-cloud-starter-appconfiguration-config/) to load configuration properties from Azure Configuration Service to Spring Environment.
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-appconfiguration-sample/README.md:3]: Seems like we shouldn't use relative links here.

https://github.com/Azure/azure-sdk/blob/master/docs/policies/README-TEMPLATE.md#link-guidelines

https://github.com/Azure/azure-sdk-for-java/pull/13989/files#diff-bbc5dba8ff02aedfeed94fd0f102c996

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-appconfiguration-sample/README.md [hunk 2]
```diff
@@ -51,7 +51,7 @@ mvn -Dspring.application.name=foo -Dspring.profiles.active=dev spring-boot:run
 
 ### More details
 
-Please refer to this [README](../../azure-spring-cloud-starter-appconfiguration-config/) about more usage details. 
+Please refer to this [README](../../../appconfiguration/azure-spring-cloud-starter-appconfiguration-config/) about more usage details.
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-appconfiguration-sample/README.md:54]: Same here.

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-feature-management-sample/README.md [hunk 3]
```diff
@@ -58,7 +58,7 @@ $ mvn -Dspring.application.name=foo -Dspring.profiles.active=dev spring-boot:run
 
 ### More details
 
-Please refer to this [README](../../azure-spring-cloud-starter-appconfiguration-config/) about more usage details. 
+Please refer to this [README](../../../appconfiguration/azure-spring-cloud-starter-appconfiguration-config/) about more usage details.
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-feature-management-sample/README.md:61]: Same here.

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-feature-management-web-sample/README.md [hunk 4]
```diff
@@ -63,7 +63,7 @@ $ mvn -Dspring.application.name=foo -Dspring.profiles.active=dev spring-boot:run
 
 ### More details
 
-Please refer to this [README](../../azure-spring-cloud-starter-appconfiguration-config/) about more usage details. 
+Please refer to this [README](../../../appconfiguration/azure-spring-cloud-starter-appconfiguration-config/) about more usage details.
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-feature-management-web-sample/README.md:66]: Same here.

**Review by chenrujun — CHANGES_REQUESTED** [review]
We shouldn't use relative links.

https://github.com/Azure/azure-sdk/blob/master/docs/policies/README-TEMPLATE.md#link-guidelines

https://github.com/Azure/azure-sdk-for-java/pull/13989/files#diff-bbc5dba8ff02aedfeed94fd0f102c996

**Review by chenrujun — APPROVED** [review]
I'll approve this PR and merge it.
Please create another PR to fix the issue about link.

**Comment by lu-cheng** [comment]
Fix the issue ablut link：https://github.com/Azure/azure-sdk-for-java/pull/14901

**Pull request merged**